src/execution_plan/optimizations/optimize_cartesian_product.c

```diff
@@ -81,13 +81,13 @@ static void _optimize_cartesian_product(ExecutionPlan *plan, OpBase *cp) {
 		// Try to create a cartesian product, followed by the current filter.
 		OpFilter *filter_op = filter_ctx_arr[i].filter;
 		OpBase **solving_branches = _find_entities_solving_branches(filter_ctx_arr[i].entities, cp);
-		if(solving_branches == NULL) {
+		uint solving_branch_count = array_len(solving_branches);
+		if(solving_branches == NULL || solving_branch_count == 0) {
```

Comment:
I think it would be better to either return NULL or a valid pointer to an array of length > 0.

Reply:
That's what we do, actually - I think the changes i made to this file were unnecessary. Reverting!